From dc3356c6859d4fb84a5e3c052f03e1848215135a Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:02:17 -0400
Subject: [fs] btrfs: remove unnecessary lock in may_commit_transaction()

Message-id: <1407179109-13714-251-git-send-email-sandeen@redhat.com>
Patchwork-id: 87193
O-Subject: [RHEL7.1 PATCH 250/422] Btrfs: remove unnecessary lock in may_commit_transaction()
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Miao Xie <miaox@cn.fujitsu.com>

commit 0424c548976b4c2a72c0bdbea425cf9d51e82d0f
Author: Miao Xie <miaox@cn.fujitsu.com>
Date:   Thu Mar 6 13:54:59 2014 +0800

    Btrfs: remove unnecessary lock in may_commit_transaction()

    The reason is:
    - The per-cpu counter has its own lock to protect itself.
    - Here we needn't get a exact value.

    Signed-off-by: Miao Xie <miaox@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent-tree.c b/fs/btrfs/extent-tree.c
index 6b821c6..5608b4f 100644
--- a/fs/btrfs/extent-tree.c
+++ b/fs/btrfs/extent-tree.c
@@ -4112,13 +4112,9 @@ static int may_commit_transaction(struct btrfs_root *root,
   goto commit;
 
  /* See if there is enough pinned space to make this reservation */
- spin_lock(&space_info->lock);
  if (percpu_counter_compare(&space_info->total_bytes_pinned,
-       bytes) >= 0) {
-  spin_unlock(&space_info->lock);
+       bytes) >= 0)
   goto commit;
- }
- spin_unlock(&space_info->lock);
 
  /*
   * See if there is some space in the delayed insertion reservation for
@@ -4127,16 +4123,13 @@ static int may_commit_transaction(struct btrfs_root *root,
  if (space_info != delayed_rsv->space_info)
   return -ENOSPC;
 
- spin_lock(&space_info->lock);
  spin_lock(&delayed_rsv->lock);
  if (percpu_counter_compare(&space_info->total_bytes_pinned,
        bytes - delayed_rsv->size) >= 0) {
   spin_unlock(&delayed_rsv->lock);
-  spin_unlock(&space_info->lock);
   return -ENOSPC;
  }
  spin_unlock(&delayed_rsv->lock);
- spin_unlock(&space_info->lock);
 
 commit:
  trans = btrfs_join_transaction(root);
-- 
1.7.1